pkg/api/aim/encoding/encoder.go

```diff
@@ -42,7 +42,7 @@ func encodeTree(w io.Writer, v any, p []any) error {
 			return err
 		}
 		for i := 0; i < rv.Len(); i++ {
-			v := rv.Index(i).Interface()
+			v = rv.Index(i).Interface()
```

Comment:
if this was to fix the linting issues for shadowing variables, I'd rather we give it a new name. It feels dangerous to me to assign something to this top-level variable when it's clearly not our intention.

Reply:
hm this is a bit strange. you are right. reverted.